indico/core/limiter.py

```diff
@@ -30,37 +30,65 @@ def __init__(self, limiter, key_func, scope, limits):
         self.scope = scope
         self.limits = limits
 
-    def _get_args(self):
-        return self.key_func(), self.scope
-
-    def hit(self):
+    def _get_args(self, args):
+        # scope args go directly the redis keys and are separated by slashes,
+        # so we want to avoid cases where there's some leakage between different
+        # rate limiters if someone ends up passing user-provided args (even though
+        # this would be a bad idea in general)
+        args = [x.replace('/', '-') if isinstance(x, str) else x for x in args]
+        if not self.key_func:
+            return self.scope, *args
+        return self.key_func(), self.scope, *args
+
+    def hit(self, *args):
         """Check the rate limit and increment the counter.
 
         This method should be used when performing an action that should
         count against the user's rate limit.
+
+        :param args: Additional args to include in the scope, such as a user id.
         """
         if self.limits is None:
             return True
-        args = self._get_args()
+        args = self._get_args(args)
         return any(self.limiter.limiter.hit(lim, *args) for lim in self.limits)
 
-    def test(self):
+    def test(self, *args):
         """Check the rate limit without incrementing the counter.
 
         This method should be used when you just want to see if the rate limit
         has been triggered (e.g. to show a message when loading a form), without
         counting the action against the rate limit.
+
+        :param args: Additional args to include in the scope, such as a user id.
         """
         if self.limits is None:
             return True
-        args = self._get_args()
+        args = self._get_args(args)
         return any(self.limiter.limiter.test(lim, *args) for lim in self.limits)
 
-    def get_reset_delay(self):
-        """Get the duration until the rate limit resets."""
+    def clear(self, *args):
+        """Reset the rate limit.
+
+        This method should be used when you want to reset any previous hits that
+        may still be affecting the limit.
+
+        :param args: Additional args to include in the scope, such as a user id.
+        """
+        if self.limits is None:
+            return True
```

Comment:
If you want to keep the API equivalent to `limiter`, the method `clear()` always returns `None`.